gradoop-data-integration/src/test/java/org/gradoop/dataintegration/transformation/ExtractPropertyTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.dataintegration.transformation;
+
+import org.apache.flink.api.java.io.LocalCollectionOutputFormat;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.common.model.impl.pojo.Edge;
+import org.gradoop.common.model.impl.pojo.Vertex;
+import org.gradoop.dataintegration.transformation.config.EdgeDirection;
+import org.gradoop.flink.model.GradoopFlinkTestBase;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.model.api.operators.UnaryGraphToGraphOperator;
+import org.gradoop.flink.model.impl.functions.epgm.ByLabel;
+import org.gradoop.flink.model.impl.functions.filters.Or;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.*;
+
+/**
+ * Tests for the {@link ExtractProperty} operator.
+ */
+public class ExtractPropertyTest extends GradoopFlinkTestBase {
+
+  /**
+   * Tests whether vertices are correctly deduplicated/condensed.
+   * @throws Exception If the data could not be loaded or collected properly.
+   */
+  @Test
+  public void cityDeduplicationTest() throws Exception {
+    LogicalGraph social = getSocialNetworkLoader().getLogicalGraph();
+
+    // this operation should create 3 vertices with the property values: Berlin, Dresden, Leipzig
+    // keep in mind that the default for condensation is 'true'
+    UnaryGraphToGraphOperator extOne = new ExtractProperty("Person", "city",
+        "City", "name");
+
+    LogicalGraph extOneGraph = social.callForGraph(extOne);
+    List<Vertex> city = extOneGraph.getVerticesByLabel("City").collect();
+    Set<String> cityNames = new HashSet<>();
```

Comment:
You are right. But IMO id would be enough to check the boolean returned by cityNames.add to ensure that no city name was added twice since this would be a bug.